File: javascript/rna-transcription/rna-transcription.js
/*jshint esversion: 6 */
var DnaTranscriber = function() {};

DnaTranscriber.prototype.toRna = function(seq) {
  'use strict';

  var sub = { 
    'G': 'C',
    'C': 'G',
    'T': 'A',
    'A': 'U'
  };

  return seq.split('').map( x => sub[x] ).join('');
};

module.exports = DnaTranscriber;

File: python/rna-transcription/rna_transcription.py
def to_rna(s):
    m = {'G': 'C', 'C': 'G', 'T': 'A', 'A': 'U'}
    r = "".join(map(lambda x: m.get(x, ''), s))
    return r if len(r) == len(s) else ''
